parsec/core/gui/users_widget.py

```diff
@@ -211,17 +211,20 @@ async def _do_list_users_and_invitations(
     omit_invitation: bool = False,
 ) -> tuple[int, list[UserInfo], list[InviteListItem]]:
     try:
-        if not pattern:
-            users, total = await core.find_humans(
-                page=page, per_page=USERS_PER_PAGE, omit_revoked=omit_revoked
-            )
-            invitations = [] if omit_invitation else await core.list_invitations()
-            return total, users, [inv for inv in invitations if inv.type == InvitationType.USER]
-        else:
-            users, total = await core.find_humans(
-                page=page, per_page=USERS_PER_PAGE, query=pattern, omit_revoked=omit_revoked
-            )
-            return total, users, []
+        invitations = [] if omit_invitation else await core.list_invitations()
+        users, total = await core.find_humans(
+            page=page, per_page=USERS_PER_PAGE, query=pattern, omit_revoked=omit_revoked
+        )
+        return (
+            total,
+            users,
+            [
+                inv
+                for inv in invitations
+                if inv.type == InvitationType.USER
+                and (pattern is not None and pattern in inv.claimer_email)
```

Comment:
What if pattern is None? Nothing is included then?

Reply:
```suggestion
                and (pattern is None or pattern in inv.claimer_email)
```

Reply:
still I would do the filtering line 214 instead

Reply:
I'm dumb. The funny thing is, it did work properly because the pattern is always provided (never actually None) as an empty string. It's fixed and I moved the filtering to line 214.